.github/workflows/node.yaml

```diff
@@ -80,87 +80,82 @@ jobs:
-            echo images $IMAGES
-            echo ::set-output name=images::"$IMAGES"
-            echo ::set-output name=dockerhub-publish::"$DOCKERHUB_PUBLISH"
+            DOCKERHUB_PUBLISH="1"
           else
-            echo "Skipping docker build"
+            DOCKERHUB_PUBLISH="0"
           fi
+          # debug output
+          echo dockerhub-publish $DOCKERHUB_PUBLISH
+          echo ::set-output name=dockerhub-publish::"$DOCKERHUB_PUBLISH"
       - name: Use Node.js
         uses: actions/setup-node@v3
-        if: ${{ steps.image-tags.outputs.images }}
         with:
           node-version-file: ".node-version"
       - uses: ./.github/actions/setup-meteor
-        if: ${{ steps.image-tags.outputs.images }}
       - name: Prepare Environment
-        if: ${{ steps.image-tags.outputs.images }}
         run: |
           yarn install
       - name: Build libs
-        if: ${{ steps.image-tags.outputs.images }}
         run: |
           yarn build:packages
       - name: Persist Built Version information
-        if: ${{ steps.image-tags.outputs.images }}
         run: |
           cd meteor
           yarn inject-git-hash
       - name: Meteor Build
-        if: ${{ steps.image-tags.outputs.images }}
         run: |
           cd meteor
           NODE_OPTIONS="--max-old-space-size=4096" METEOR_DEBUG_BUILD=1 meteor build --allow-superuser --directory .
       - name: Meteor Bundle NPM Build
-        if: ${{ steps.image-tags.outputs.images }}
         run: |
           cd meteor/bundle/programs/server
           meteor npm install
       - name: Set up Docker Buildx
-        if: ${{ steps.image-tags.outputs.images }}
         uses: docker/setup-buildx-action@v2
       - name: Login to DockerHub
-        if: steps.image-tags.outputs.images && steps.image-tags.outputs.dockerhub-publish == '1'
+        if: steps.dockerhub.outputs.dockerhub-publish == '1'
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      # TODO - do we want this?
-      # - name: Login to GitHub Container Registry
-      #   uses: docker/login-action@v2
-      #   with:
-      #     registry: ghcr.io
-      #     username: ${{ github.repository_owner }}
-      #     password: ${{ secrets.CR_PAT }}
-      - name: Build and push
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push to GHCR
         uses: docker/build-push-action@v3
-        if: ${{ steps.image-tags.outputs.images }}
         with:
           context: .
           file: ./meteor/Dockerfile.circle
           push: true
-          tags: ${{ steps.image-tags.outputs.images }}
+          tags: "ghcr.io/nrkno/${{ steps.docker-tag.outputs.tags }}"
```

Comment:
> Is there anyone at tv2 who can weigh in on weather or not this is a problem for them?

Perhaps it will be sufficient to use `"ghcr.io/${{ github.repository_owner }}/${{ steps.docker-tag.outputs.tags }}"`. That way it might publish to ghcr for any fork.

I'm not sure who to ping, but a message on the shared slack should get a response from the right people.

> Should it be sofie-core/<server-core|playout-gateway|mos-gateway> for ghcr?

Yeah I'm happy with that

Reply:
Looks like what you're suggesting @Julusian will work, but It's unclear to me how to opt out of publishing to GitHub Container Registry, and to only publish to Dockerhub

Reply:
Ok, let me know if I've understood the situation correctly. You publish images to DockerHub from your fork and your prefix is stored in `secrets.DOCKERHUB_IMAGE_PREFIX`? Then I'll have to use this secret to make up the image name for pushing to DockerHub.

I think we could use a secret for enabling/disabling push to GHCR, unless any of you can think of a better method?

Reply:
@siljekristensen We're using the `secrets.DOCKERHUB_IMAGE_PREFIX`, and I believe it must be like `tv2media/sofie-`, because then [here](https://github.com/nrkno/sofie-core/pull/803/files#diff-9343bec002b600fecb6e4f230466c2b3d5c3f9eb64712f3d1f888e3fa0a98e05L100) we used to end up with `tv2media/sofie-server-core`, whereas in the gateways it becomes for example `tv2media/sofie-playout-gateway`. I'm not sure what that prefix would have to be after this change, in order to work [here](https://github.com/nrkno/sofie-core/pull/803/files#diff-9343bec002b600fecb6e4f230466c2b3d5c3f9eb64712f3d1f888e3fa0a98e05R183) as well as [here](https://github.com/nrkno/sofie-core/blob/51aef59371dea53177eeb191077745dc8b21d866/.github/workflows/node.yaml#L218)

A secret for enabling/disabling push to GHCR sounds good to me.

Reply:
Alright, so I've added a secret called `GHCR_ENABLED`. The if statements are a bit much now, but it should ensure that all the build steps aren't done when GHCR isn't enabled and push to DockerHub isn't enabled ([example run](https://github.com/nrkno/sofie-core/actions/runs/3538974288/jobs/5940361510)).

Does this work for everyone? @ianshade @Julusian

Reply:
If you put the `check-build-and-push` after `dockerhub` and `check-ghcr`, then you could do the `(steps.dockerhub.outputs.dockerhub-publish == '1' || steps.check-ghcr.outputs.enable == 'true')` check in that step, and not have to repeat it everywhere.

Or perhaps those 3 could be combined into one step, as they are all doing some simple bash checks?

Im happy with this with or without these suggestions :)

Reply:
That's way nicer, I've changed it and also updated the gateway build job :wrench: